tests/utils/iotrace.py

```diff
@@ -120,132 +188,288 @@ def get_sort_key(element):
-
-        :return: List of dictionaries with fields corresponding to json messages fields
-        :rtype: List of dicts
-        '''
-        msgs_list = []
-        for obj in self.__decode_json_stream(output):
-            msgs_list.append(obj)
-        return msgs_list
-
-    # Get json objects from a stream of concatenated json messages
-    #
-    # raw_decode stops once it has a valid object and returns the last position
-    # where wasn't part of the parsed object. It's not documented, but you can
-    # pass this position back to raw_decode and it start parsing again from that
-    # position. Unfortunately, the Python json module doesn't accept strings
-    # that have prefixing whitespace. So we need to search to find the first
-    # none-whitespace part of your document.
-    def __decode_json_stream(self, document, pos=0, decoder=json.JSONDecoder()):
-        while True:
-            # Create json stream without whitespace
-            match = NOT_WHITESPACE.search(document, pos)
-
-            if not match:
-                # No more data
-                return
-            pos = match.start()
-
-            try:
-                obj, pos = decoder.raw_decode(document, pos)
-            except json.JSONDecodeError:
-                raise Exception("Invalid json formatting")
-
-            yield obj
+        :param shortcut: Use shorter command
+        :type trace_path: str
+        :type shortcut: bool
+        :raises Exception: if setting path fails
+        """
+        command = 'iotrace' + (' -C' if shortcut else ' --trace-config')
+        command += ' -S ' if shortcut else ' --set-trace-repository-path '
+        command += (' -p ' if shortcut else ' --path ') + f'{trace_path}'
+        error_output = parse_json(
+            TestRun.executor.run(command).stderr)[0]["trace"]
+        if error_output == "No access to trace directory":
+            raise CmdException("Invalid setting of the trace repository path")
+
+    @staticmethod
+    def help(shortcut: bool = False) -> str:
+        """
+        :param shortcut: Use shorter command
+        :type shortcut: bool
+        :return: io-tracer help
+        :raises Exception: if command fails
+        """
+        return TestRun.executor.run_expect_success(
+            'iotrace' + (' -H ' if shortcut else ' --help ')).stdout
+
+    @staticmethod
+    def version(shortcut: bool = False) -> list:
+        """
+        :param shortcut: Use shorter command
+        :type shortcut: bool
+        :return: io-tracer version
+        :rtype: list of dictionaries
+        :raises Exception: if command fails
+        """
+        output = parse_json(TestRun.executor.run_expect_success(
+            'iotrace' + (' -V ' if shortcut else ' --version ')))
+        version = (output[0]["system"] + " version: " + output[0]["trace"]
+                   + "\n" + output[1]["system"] + " version: "
+                   + output[1]["trace"] + "\n")
+        return version
```

Comment:
Doesn't this return a string?
Also I'd drop the custom display format, and either return 'output'
or return the string with 'output[1]["trace"]' - to make parsing this return more intuitive

Reply:
I decided to return a dictionary, it would be more readable and also easy to parse.